### PR TITLE
Transpile MyApp.java

### PR DESCRIPTION
**Pull request by isiko**
# Description
- Transpiled the main Application Class from Java to Kotlin.

# Fully tested on (with all Build flavors):
- Tablet (Pixel C), API lvl 24 Emulator
- Tablet (Pixel C), API lvl 30 Emulator
- Phone (Pixel 5), API lvl 31 Emulator
- Phone (Pixel 2), API lvl 24 Emulator

### CODE REVIEW

**Comment by isiko** [comment]
Also, this might be an opportunity to rename this class, many to something like `EventFahrplanApp` or something.

**Comment by johnjohndoe** [comment]
I would just name the class `App`.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            BuildConfig.SCHEDULE_LAST_DAY_END_DAY
+        )
+
+        @JvmField
+        val conferenceTimeFrame = ConferenceTimeFrame(FIRST_DAY_START, LAST_DAY_END)
+
+        @JvmField
+        var task_running = TASKS.NONE
+
+        private fun getMilliseconds(
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:57]: A bit more concise:

``` kotlin
private fun getMilliseconds(timeZoneId: String, year: Int, month: Int, day: Int): Long {
    val zeroBasedMonth = month - 1
    val zone = TimeZone.getTimeZone(timeZoneId)
    return GregorianCalendar(zone).apply {
        set(year, zeroBasedMonth, day, 0, 0, 0)
        set(Calendar.MILLISECOND, 0)
    }.timeInMillis
}
```

Please move the helper function under the properties where it is used.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    enum class TASKS {
+        NONE, FETCH, PARSE, FETCH_CANCELLED
+    }
+
+    @CallSuper
+    override fun onCreate() {
+        super.onCreate()
+        TraceDroid.init(this)
+        task_running = TASKS.NONE
+        initialize(
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:27]: Please use the class names so one can understand what happens here:

``` kotlin
AppRepository.initialize(
    applicationContext,
    Logging.get()
)
```

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.*
+
+import info.metadude.android.eventfahrplan.commons.logging.Logging.Companion.get
+import nerd.tuxmobil.fahrplan.congress.repositories.AppRepository.initialize
+import nerd.tuxmobil.fahrplan.congress.utils.ConferenceTimeFrame
+
+class MyApp : Application() {
+
+    enum class TASKS {
+        NONE, FETCH, PARSE, FETCH_CANCELLED
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:19]: `FETCH_CANCELLED` seems to be used. Please check the Git history when this happened and mention it in your commit message.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        task_running = TASKS.NONE
+        initialize(
+            applicationContext,
+            get()
+        )
+    }
+
+    companion object {
+
+        const val DEBUG = false
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:35]: `DEBUG` can be private.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        super.onCreate()
+        TraceDroid.init(this)
+        task_running = TASKS.NONE
+        initialize(
+            applicationContext,
+            get()
+        )
+    }
+
+    companion object {
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:33]: Please move the `companion object` above `onCreate()`.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
+package nerd.tuxmobil.fahrplan.congress
+
+import android.app.Application
+import android.util.Log
+
+import androidx.annotation.CallSuper
+
+import org.ligi.tracedroid.TraceDroid
+
+import java.util.*
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:10]: No `*` imports please.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+            BuildConfig.SCHEDULE_LAST_DAY_END_YEAR,
+            BuildConfig.SCHEDULE_LAST_DAY_END_MONTH,
+            BuildConfig.SCHEDULE_LAST_DAY_END_DAY
+        )
+
+        @JvmField
+        val conferenceTimeFrame = ConferenceTimeFrame(FIRST_DAY_START, LAST_DAY_END)
+
+        @JvmField
+        var task_running = TASKS.NONE
```
Comment (johnjohndoe) [inline app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:55]: Please use camelCase.

**Review by johnjohndoe — CHANGES_REQUESTED** [review]
Here is a few improvements for the application class. If you still want to rename it I suggest doing this in a separate commit because Git can easier deal with it when there are no additional changes.

**Comment by isiko** [comment]
> If you still want to rename it...

I'll do, but wanted to wait for the other pull requests to avoid merging issues 😅

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt (hunk 3 above)
Comment (isiko) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/MyApp.kt:19]: Actually, it seems like it was never used 😄 I searched a bit but just found that it was added in 79e35f13 and reformated in 2cac2a38 but wasn't used anywhere 🤔
I'll remove it now.

**Review by johnjohndoe — APPROVED** [review]
Ready to merge. After rebasing your branch one JVM annotation became redundant - so I removed it. I also re-organized the imports.

**Pull request merged**